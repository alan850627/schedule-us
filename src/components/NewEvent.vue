<template>
  <div class="new-event">
    <h1>Create a new event!</h1>
    <button v-on:click="submit()">Submit</button>
  </div>
</template>

<script>
import db from '../firebaseInit'

export default {
  name: 'NewEvent',

  components: {
  },

  props: {
  },

  computed: {
  },

  data () {
    return {
    }
  },

  methods: {
    submit: function () {
      let event = {
        time: Date.now()
      }
      db.collection('events').doc().set(event)
    },

    // May move/update this.
    newChatBox: function () {
      let chatBox = {}
      chatBox[`${Date.now()}`] = {
        username: 'System',
        message: 'Chat created.'
      }

      db.collection('chat-boxes').doc().set(chatBox)
    }
  },

  mounted () {
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

</style>
