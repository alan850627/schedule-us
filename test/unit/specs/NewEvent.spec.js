import NewEvent from '@/components/NewEvent'
import { shallow } from 'vue-test-utils'

describe('NewEvent.vue', () => {
  it('has the expected html structure', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    expect(cmp.element).toMatchSnapshot()
    expect(cmp.vm).toBeDefined()
  })
  it('submit', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.submit()
    expect(cmp.vm).toBeDefined()
  })
  it('submit', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.submit()
    expect(cmp.vm).toBeDefined()
  })
  it('reset', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.hostEmail = 'alan@gmail.com'
    cmp.vm.hostName = 'alan'
    cmp.vm.description = 'yes'
    cmp.vm.eventName = 'yes'
    cmp.vm.start = '100000000'
    cmp.vm.reset()
    expect(cmp.vm.hostEmail).toBe('')
  })
  it('makeChatBox', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    expect(cmp.vm.makeChatBox()).toBeDefined()
  })
  it('makeTimeTable', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    expect(cmp.vm.makeTimeTable(Date.now())).toBeDefined()
  })
  it('username change', () => {
    let cmp = shallow(NewEvent, {
      propsData: {
        username: 'test-user'
      }
    })
    cmp.vm.username = 'Alan'
    cmp.vm.$nextTick(() => {
      expect(cmp.vm.hostName).toBe('Alan')
    })
  })
})
