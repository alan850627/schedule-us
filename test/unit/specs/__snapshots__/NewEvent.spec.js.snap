// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewEvent.vue has the expected html structure 1`] = `
<b-container
  class="new-event"
>
  <b-row
    align-h="center"
  >
    <b-col
      cols="8"
    >
      <b-card>
        <h1
          style="text-align: center;"
        >
           Create a new event!
        </h1>
         
        <br />
         
        <b-form-group
          id=""
          label="Event Name:"
          label-for="eventNameForm"
        >
          <b-form-input
            id="eventNameForm"
            required=""
            type="text"
          />
        </b-form-group>
         
        <b-form-group
          id=""
          label="Event Description:"
          label-for="eventDescriptionForm"
        >
          <b-form-input
            id="eventDescriptionForm"
            type="text"
          />
        </b-form-group>
         
        <b-form-group
          id=""
          label="Possible Event Start Date:"
          label-for="eventStartDateForm"
        >
          <b-form-input
            id="eventStartDateForm"
            required=""
            type="date"
          />
        </b-form-group>
         
        <b-form-group
          id=""
          label="Possible Event End Date:"
          label-for="eventEndDateForm"
        >
          <b-form-input
            id="eventEndDateForm"
            required=""
            type="date"
          />
        </b-form-group>
         
        <b-form-group
          id=""
          label="Your Name:"
          label-for="exampleInput3"
        >
          <b-form-input
            id="exampleInput3"
            type="text"
          />
        </b-form-group>
         
        <b-form-group
          id=""
          label="Email:"
          label-for="exampleInput4"
        >
          <b-form-input
            id="exampleInput4"
            type="email"
          />
        </b-form-group>
         
        <b-button
          variant="primary"
        >
          Submit
        </b-button>
         
        <b-button
          variant="danger"
        >
          Reset
        </b-button>
      </b-card>
    </b-col>
  </b-row>
</b-container>
`;
